test/functional/wallet_encryption.py

```diff
@@ -46,6 +48,25 @@ def run_test(self):
         assert_raises_rpc_error(-8, "passphrase cannot be empty", self.nodes[0].walletpassphrase, '', 1)
         assert_raises_rpc_error(-8, "passphrase cannot be empty", self.nodes[0].walletpassphrasechange, '', 'ff')
 
+        # Make sure there are no unencrypted key records
+        # Use the wallet tool's dump mechanism to get all of the records to inspect them.
+        self.nodes[0].unloadwallet(self.default_wallet_name)
+        dump_file = os.path.join(self.nodes[0].datadir, "wallet.dump")
+        subprocess.check_call([self.options.bitcoinwallet, f"-datadir={self.nodes[0].datadir}", f"-chain={self.chain}", f"-wallet={self.default_wallet_name}", f"-dumpfile={dump_file}", "dump"], stdout=subprocess.DEVNULL)
```

Comment:
In e986d5a15:

Same here, need to change `f"-datadir={self.nodes[0].datadir}"` for `f"-datadir={self.nodes[0].datadir_path}"`.

```suggestion
        subprocess.check_call([self.options.bitcoinwallet, f"-datadir={self.nodes[0].datadir_path}", f"-chain={self.chain}", f"-wallet={self.default_wallet_name}", f"-dumpfile={dump_file}", "dump"], stdout=subprocess.DEVNULL)
```

Reply:
Fixed